src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -66,12 +76,57 @@ RouterExecutorStart(QueryDesc *queryDesc, int eflags, Task *task)
 	/* ensure that the task is not NULL */
 	Assert(task != NULL);
 
-	/* disallow transactions and triggers during distributed modify commands */
+	/* disallow triggers during distributed modify commands */
 	if (commandType != CMD_SELECT)
 	{
-		bool topLevel = true;
-		PreventTransactionChain(topLevel, "distributed commands");
 		eflags |= EXEC_FLAG_SKIP_TRIGGERS;
+
+		if (IsTransactionBlock())
```

Comment:
Chatted with @anarazel on chat about this. He said he misremembered and that detecting multi-statement/function contexts for non-utility statements has no trivial solution, so we're skipping this for now:

> Yea, there's no easy method. So let's skip that for now.